src/main/java/org/zalando/nakadi/repository/db/EventTypeCache.java

```diff
@@ -235,6 +245,10 @@ private String getZNodePath(final String eventTypeName) {
         return ZKPaths.makePath(ZKNODE_PATH, eventTypeName);
     }
 
+    public void addInvalidationListener(final Consumer<String> onEventTypeInvalidated) {
```

Comment:
If you call method, while iterating over it in line 223, you got exception. Also you access arraylist from different threads ...